iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/BusBreakerVoltageLevelAdapter.java

```diff
@@ -0,0 +1,211 @@
+
+        @Override
+        public int getSwitchCount() {
+            return getDelegate().getSwitchCount();
+        }
+
+        @Override
+        public void removeSwitch(String switchId) {
+            // TODO(mathbagu)
+            throw MergingView.createNotImplementedException();
+        }
+
+        @Override
+        public void removeAllSwitches() {
+            // TODO(mathbagu)
+            throw MergingView.createNotImplementedException();
+        }
+
+        @Override
+        public Bus getBus1(String switchId) {
+            return getBus(getDelegate().getBus1(switchId));
+        }
+
+        @Override
+        public Bus getBus2(String switchId) {
+            return getBus(getDelegate().getBus2(switchId));
+        }
+
+        @Override
+        public Switch getSwitch(String switchId) {
+            return getIndex().getSwitch(getDelegate().getSwitch(switchId));
+        }
+
+        @Override
+        public SwitchAdder newSwitch() {
+            return new SwitchAdderAdapter(getDelegate().newSwitch(), getIndex());
+        }
+
+        @Override
+        public Bus getBus(Bus bus) {
+            return getIndex().getBus(bus);
+        }
+    }
+
+    static class BusViewAdapter extends BusCache<BusView> implements BusViewExt {
+
+        BusViewAdapter(BusView delegate, MergingViewIndex index) {
+            super(delegate, index);
+        }
+
+        @Override
+        public Iterable<Bus> getBuses() {
+            return Iterables.transform(getDelegate().getBuses(), this::getBus);
+        }
+
+        @Override
+        public Stream<Bus> getBusStream() {
+            return getDelegate().getBusStream().map(this::getBus);
+        }
+
+        @Override
+        public Bus getBus(String id) {
+            return getBus(getDelegate().getBus(id));
+        }
+
+        @Override
+        public Bus getMergedBus(String configuredBusId) {
+            return getBus(getDelegate().getMergedBus(configuredBusId));
+        }
+    }
+
+    private final BusBreakerViewAdapter busBreakerView;
+
+    private final BusViewAdapter busView;
+
+    BusBreakerVoltageLevelAdapter(final VoltageLevel delegate, final MergingViewIndex index) {
+        super(delegate, index);
+        busBreakerView = new BusBreakerViewAdapter(delegate.getBusBreakerView(), index);
+        busView = new BusViewAdapter(delegate.getBusView(), index);
+    }
+
+    @Override
+    public NodeBreakerView getNodeBreakerView() {
+        // The delegate is a BusBreakerVoltageLevel, so we can return its Node/Breaker view that throws exception for all the methods.
+        return getDelegate().getNodeBreakerView();
+    }
+
+    @Override
+    public BusBreakerViewExt getBusBreakerView() {
+        return busBreakerView;
+    }
+
+    @Override
+    public BusViewExt getBusView() {
+        return busView;
+    }
+
+    @Override
+    public void invalidateCache() {
+        busView.invalidateCache();
+        System.out.println("BBKVLA::invalidate");
```

Comment:
Is this `System.out.println` voluntary?

Reply:
Just to debug, but you right, I have to remove it